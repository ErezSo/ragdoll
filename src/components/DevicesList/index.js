import React from "react";
import { Device } from "../index";

const styles = {
  container: {
    width: "100%",
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center"
  }
};

const DevicesList = ({ devices, searchTerm, toggleDevice }) => {
  const { container } = styles;
  return (
    <div style={container}>
      {devices
        .filter(
          device =>
            device.name.toUpperCase().indexOf(searchTerm.toUpperCase()) >= 0
        )
        .map(device => (
          <Device
            key={device.timestamp}
            device={device}
            toggleDevice={toggleDevice}
          />
        ))}
    </div>
  );
};

export default DevicesList;
