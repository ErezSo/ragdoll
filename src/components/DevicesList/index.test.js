import React from "react";
import { shallow } from "enzyme";
import { Device } from "../index";
import DevicesList from "./";

describe("DevicesList", () => {
  let wrapper;
  const devices = [
    {
      name: "rotation_rate_alpha",
      unit: "deg/s",
      value: 0,
      timestamp: 1528878066606,
      active: false
    },
    {
      name: "rotation_rate_gamma",
      unit: "deg/s",
      value: 0,
      timestamp: 1528878066506,
      active: true
    }
  ];

  beforeEach(() => {
    wrapper = shallow(<DevicesList devices={devices} searchTerm="" />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("render only items that match the search term", () => {
    wrapper = shallow(<DevicesList devices={devices} searchTerm="alpha" />);
    expect(wrapper.find(Device)).toHaveLength(1);
  });
});
