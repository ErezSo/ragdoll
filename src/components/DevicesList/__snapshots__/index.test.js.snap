// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevicesList renders correctly 1`] = `
"<styled.div>
  <Device device={{...}} toggleDevice={[undefined]} />
  <Device device={{...}} toggleDevice={[undefined]} />
</styled.div>"
`;
